src/Diagnostics.RuntimeHost/Services/StorageService/StorageService.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Diagnostics.ModelsAndUtils.Models.Storage;
+using Microsoft.Extensions.Configuration;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Table;
+using Diagnostics.Logger;
+using Diagnostics.RuntimeHost.Utilities;
+using Microsoft.AspNetCore.Hosting;
+namespace Diagnostics.RuntimeHost.Services.StorageService
+{
+    public interface IStorageService
+    {
+        bool GetStorageFlag();
+        Task<List<DiagEntity>> RetieveEntitiesByPartitionkey(string partitionKey = null);
+        Task<DiagEntity> LoadDataToTable(DiagEntity detectorEntity);
+    }
+    public class StorageService : IStorageService
+    {
+        public static readonly string PartitionKey = "PartitionKey";
+        public static readonly string RowKey = "RowKey";
+        
+        private static CloudTableClient tableClient;
+
+        private string tableName;
+        private bool loadOnlyPublicDetectors;
+        private bool isStorageEnabled;
+
+        public StorageService(IConfiguration configuration, IHostingEnvironment hostingEnvironment)
+        {
+            tableName = configuration["SourceWatcher:TableName"];
+            if(hostingEnvironment != null && hostingEnvironment.EnvironmentName.Equals("UnitTest", StringComparison.CurrentCultureIgnoreCase))
+            {
+                tableClient = CloudStorageAccount.DevelopmentStorageAccount.CreateCloudTableClient();
+            } else
+            {
+                var accountname = configuration["SourceWatcher:DiagStorageAccount"];
+                var key = configuration["SourceWatcher:DiagStorageKey"];
+                var storageAccount = new CloudStorageAccount(new StorageCredentials(accountname, key), accountname, "core.windows.net", true);
+                tableClient = storageAccount.CreateCloudTableClient();
+            }
+         
+            if (!bool.TryParse((configuration[$"SourceWatcher:{RegistryConstants.LoadOnlyPublicDetectorsKey}"]), out loadOnlyPublicDetectors))
+            {
+                loadOnlyPublicDetectors = false;
+            }
+
+            if(!bool.TryParse((configuration["SourceWatcher:UseStorageAsSource"]), out isStorageEnabled))
+            {
+                isStorageEnabled = false;
+            }
+
+            CheckTable();
+        }
+
+
+        private void CheckTable()
+        {
+            CloudTable table = tableClient.GetTableReference(tableName);
+            table.CreateIfNotExistsAsync();
```

Comment:
This looks like an awaitable task inside a void method. Two possible issues here:
- This method will run sequentially (Prob not that big impact as this is only called once in Service Ctor)
- Since we are not awaiting the task, there is no guarantee that the table is created when this method exits.